Partial-CI-Tests-Structure.md

```diff
@@ -0,0 +1,22 @@
+## Introduction
+
+Our CI process has a workflow, `check_ci_test_suites_to_run`, that determines which test suites to run for a given PR based on the files that the PR modified.
+If your PR fails due to the `check_ci_test_suites_to_run` workflow, please look at the error log and depending on the error, do the following:
+
+* If you see an error about the root files config, then see the [root files config section](#root-files-config) below.
+
+## Overview of Partial CI Tests Structure
+
+For non-python files, we run partial ci tests depending on the tests that a specific file affects. Running partial tests like this allow for PRs to be checked faster on the tests that a PR actually modifies. This faster test speed allows developers to get feedback faster and allows CI resources to be free more. The process of determining which tests to run goes like this:
+
+1. Generate a root files mapping which maps files to their root files. This file is generated at the start of every PR check run and is stored at `root-files-mapping.json`.
+2. Store golden files containing the page modules that a specific test uses. These golden files are stored at `core/tests/test-modules-mappings/{{test-type}}/{{test-name}}`.
+3. Run the script, `scripts/check_ci_test_suites_to_run.py` inside the Github workflow which will map changed files in a PR to their root files using the mapping above. Afterwards, map those root files to tests by using the golden files containing the page modules that a test uses.
+
+# Root Files Config
+
+The root files config at `core/tests/root-files-config.json` contains all the valid root files that are generated in the root files mapping. This ensures that all files in the codebase either lead to a root file or are a root file themselves. We also have a list of root files that should run all tests. The valid root files are stored in the field `VALID_ROOT_FILES`, while the run all tests root files are stored in the field `RUN_ALL_TESTS_ROOT_FILES`.
```

Comment:
This is a bit confusing. The root files mapping seems to be generated at the start of each PR check run, but this root files config seems to be a static file in the codebase. Wouldn't the root files mapping differ for each PR, and if so, how can its files match the static file all the time?

Also, as a general note, you can use hyphenation or quotes to make some of the "idea boundaries" clearer, e.g. "root-files config" and `"run-all-tests root files"` or `"run all tests" root files`.

Reply:
Sorry, can you elaborate on your first paragraph. The root files config is a way to validate that all root files are captured, so if a PR adds a new root file, there should be an error as we compare the dynamic root files mapping with the root files config and if it isn't there, the developer should add it to VALID_ROOT_FILES or RUN_ALL_TESTS_ROOT_FILES.

Reply:
You say, above, "Generate a root files mapping which maps files to their root files. This file is generated at the start of every PR check run." That's what I'm referring to at the start of my first paragraph.

Then you have this "root file config", which seems to be a static file. The paragraph here says "The root files config at `core/tests/root-files-config.json` contains all the valid root files that are generated in the root files mapping." which implies that the files in root-files-mapping and root-files-config match. Did you mean to say something like "the root files config contains a list of all the valid root files, and the root files mapping generated in each PR is a subset of this list"? (Also not sure what "valid" means here -- I took it to mean "exists in the codebase" but if you had a different meaning for it like below, that might be worth clarifying too.)

Reply:
The root-files-config.json and the root files mapping should match or the check will give an error and therefore the developer should update the root-files-config.json so they do match. Valid means that it is a root file that should have no parents (README, page modules, etc.). Most of the time the root-files-config.json will contain the same valid root files as the root files mapping, as a developer isn't adding any new pages or files (that aren't referenced by another file). 

Given this, I think I can explain more about how the root-files-mapping and root-files-config can differ and in this case it will give an error?

Reply:
Yes, I think that might help. I also wonder if it's better to find a name that's not "valid" as it's not clear to a general developer what "valid" means in this context.

Reply:
Done.